### PR TITLE
Add query building to query builder

### LINKED ISSUE #1685: Selections in the query builder should build a editable and submittable query
Depends on #1683.

- Selections in the visual query builder should build a query behind the scenes that can be submitted to view data
- ~~A user should be able edit this query as text (Flux or InfluxQL)~~ #1775
- ~~If the user selects a function not supported in InfluxQL, the ability to edit the query as InfluxQL should be disabled~~ #1775
- The user should need to press the submit button explicitly
- The submit button should be disabled if the query is not fully built (e.g. no bucket selected)

### PR DESCRIPTION
**Pull request by chnn**
Closes #1685

builder

### CODE REVIEW

File: ui/src/shared/components/SubmitQueryButton.tsx [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+        text="Submit"
+        size={ComponentSize.Small}
+        status={this.buttonStatus}
+        onClick={this.handleClick}
+        color={ComponentColor.Primary}
+      />
+    )
+  }
+
+  private get buttonStatus(): ComponentStatus {
```
Comment (AlirieGray) [inline ui/src/shared/components/SubmitQueryButton.tsx]: nice

File: ui/src/shared/utils/queryBuilder.test.ts [hunk 2]
```diff
@@ -0,0 +1,87 @@
+import {buildQuery} from 'src/shared/utils/queryBuilder'
+
+import {BuilderConfig} from 'src/types/v2'
+
+describe('buildQuery', () => {
```
Comment (AlirieGray) [inline ui/src/shared/utils/queryBuilder.test.ts]: yay tests!

File: ui/src/shared/utils/queryBuilder.ts [hunk 3]
```diff
@@ -0,0 +1,89 @@
+import {BuilderConfig} from 'src/types/v2'
+import {FUNCTIONS} from 'src/shared/constants/queryBuilder'
+
+const DEFAULT_WINDOW_INTERVAL = '10s'
+
+const WINDOW_INTERVALS = {
```
Comment (AlirieGray) [inline ui/src/shared/utils/queryBuilder.ts]: what does this do?

Comment (chnn) [reply ui/src/shared/utils/queryBuilder.ts]: It's a mapping of time ranges to what interval we should use when viewing that time range.

So like if we are viewing 1 hour of data, we should see it aggregated by 1 minute intervals.

It's kind of a temporary thing... I think we'll eventually use `autoInterval` or something like that.

**Pull request merged**